flink-runtime/src/main/java/org/apache/flink/runtime/util/bash/BashJavaUtils.java

```diff
@@ -40,18 +43,28 @@
 	@VisibleForTesting
 	public static final String EXECUTION_PREFIX = "BASH_JAVA_UTILS_EXEC_RESULT:";
 
-	public static void main(String[] args) throws Exception {
+	private BashJavaUtils() {
+	}
+
+	public static void main(String[] args) throws FlinkException {
 		checkArgument(args.length > 0, "Command not specified.");
 
+		Command command = Command.valueOf(args[0]);
 		String[] commandArgs = Arrays.copyOfRange(args, 1, args.length);
+		Configuration configuration = FlinkConfigLoader.loadConfiguration(commandArgs);
```

Comment:
I'm not entirely sure about this refactor.

I'm good with always take the first argument as the command, and always print the outputs with the prefix, because these are IMO common contract for all the `BashJavaUtils` commands.

However, the assumption that the remaining arguments are used and only used for generating `Configuration` may not always be true. It is only true based on the current state with the `GET_JM/TM_RESOURCE_PARAMS` commands. This does not cause any problem ATM, but I think it might be better to pass in the `commandArgs` and generate `Configuration` inside `getJm/TmResourceParams`.

Reply:
I did it to simplify unit testing of `getJm/TmResourceParams`, otherwise we would have to set config options in `flink-conf.yaml` and we do not need to test the configuration loading for this change or similar changes. I can move `FlinkConfigLoader.loadConfiguration` to `runCommand` and call `getJm/TmResourceParams` in unit tests. Then commands can use `commandArgs` later if needed.